src/server/game/Server/Packets/InstancePackets.h

```diff
@@ -164,6 +164,41 @@ namespace WorldPackets
             bool Locked = false;
             bool Extended = false;
         };
+
+        class SendEncounterEngage final : public ServerPacket
+        {
+        public:
+            SendEncounterEngage() : ServerPacket(SMSG_INSTANCE_ENCOUNTER_ENGAGE_UNIT, 15) { }
+
+            WorldPacket const* Write() override;
+
+            ObjectGuid guid;
+            uint8 priority = 0; // used to set the initial position of the frame if multiple frames are sent
+
+        };
+
+        class SendEncounterDisengage final : public ServerPacket
+        {
+        public:
+            SendEncounterDisengage() : ServerPacket(SMSG_INSTANCE_ENCOUNTER_DISENGAGE_UNIT, 15) { }
+
+            WorldPacket const* Write() override;
+
+            ObjectGuid guid;
+        };
+
+        class SendEncounterChangePriority final : public ServerPacket
+        {
+        public:
+            SendEncounterChangePriority() : ServerPacket(SMSG_INSTANCE_ENCOUNTER_CHANGE_PRIORITY, 15) { }
+
+            WorldPacket const* Write() override;
+
+            ObjectGuid guid;
+            uint8 priority = 0; // used to update the position of the unit's current frame
+
+        };
```

Comment:
class InstanceEncounterEngageUnit
    {
        ObjectGuid Unit;
        uint8 TargetFramePriority = 0;
    };

    class InstanceEncounterDisengageUnit
    {
        ObjectGuid Unit;
    };

    class InstanceEncounterChangePriority
    {
        ObjectGuid Unit;
        uint8 TargetFramePriority = 0;
    };

Please stick to these names